pygef/gef.py

```diff
@@ -588,83 +592,83 @@ def __init__(self, header_s, data_s, zid, height_system):
             .pipe(self.replace_column_void, self.column_void)
             .pipe(self.correct_pre_excavated_depth, self.pre_excavated_depth)
             .pipe(self.correct_depth_with_inclination)
-            .pipe(lambda df: df.assign(depth=np.abs(df["depth"].values)))
+            .pipe(self.make_depth_absolute)
             .pipe(self.calculate_elevation_with_respect_to_nap, zid, height_system)
             .pipe(self.calculate_friction_number)
         )
 
     @staticmethod
     def replace_column_void(df, column_void):
+        # TODO: add interpolation for none values
         if column_void is not None:
-            # added drop nan because values can't be extrapolated
-            return (
-                df.replace(column_void, np.nan)
-                .interpolate(method="linear")
-                .dropna()
-                .reset_index(drop=True)
-            )
+            # TODO: what to do with multiple columnvoids?
+            if isinstance(column_void, list):
+                df = df.filter(pl.col("*") != column_void[0])
+            else:
+                df = df.filter(pl.col("*") != column_void)
+
         return df
 
     @staticmethod
     def calculate_friction_number(df):
         if "fs" in df.columns and "qc" in df.columns:
-            df = df.assign(friction_number=(df["fs"].values / df["qc"].values * 100))
+            df["friction_number"] = df["fs"] / df["qc"] * 100
         if "friction_number" not in df.columns:
-            df = df.assign(friction_number=(np.zeros(df.shape[0])))
+            df["friction_number"] = np.zeros(df.shape[0])
```

Comment:
This should all be expressions.

`col("fs") / col("qc") * 100.0`

`pl.lit(0.0).alias("friction_number")`